File: dart/bob/lib/bob.dart
class Bob {
  // Put your code here

  String response(String sentence) {
    var trimmed = sentence.trim();

    if (trimmed.isEmpty) {
      return 'Fine. Be that way!';
    }

    bool isQuestion = trimmed[trimmed.length - 1] == '?';

    if (trimmed.toLowerCase().toUpperCase() == trimmed &&
        trimmed.contains(RegExp(r"[a-zA-Z]{1}"))) {
      if (isQuestion) {
        return 'Calm down, I know what I\'m doing!';
      }

      return 'Whoa, chill out!';
    }

    if (isQuestion) {
      return 'Sure.';
    }

    return 'Whatever.';
  }
}

File: fsharp/bob/Bob.fs
module Bob

let isYell (input: string) : bool =
    input.ToUpper() = input && input.ToLower() <> input.ToUpper()

let isQuestion (input: string) : bool =
    if input.Length > 0 then
        input[input.Length - 1] = '?'
    else
        false

let response (input: string) : string =
    let formatted = input.Trim()

    let question = isQuestion formatted
    let yell = isYell formatted

    if (formatted = "") then
        "Fine. Be that way!"
    elif (question && yell) then
        "Calm down, I know what I'm doing!"
    elif (yell) then
        "Whoa, chill out!"
    elif (question) then
        "Sure."
    else
        "Whatever."
